src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -305,6 +305,28 @@ static void write_tunnel_oper_items(private_kernel_super_ipsec_t* this,
 	LOG(DBG_KNL, SL_LOG_ERR, "Failed to write tunnel");
 }
 
+static bool get_tunnel_lifetime_intervals(ike_sa_t *ike_sa, struct terminator_tunnel_lifetime_intervals *intervals) {
+	peer_cfg_t *peer_cfg = ike_sa->get_peer_cfg(ike_sa);
+	if(! peer_cfg) {
+		LOG(DBG_KNL, SL_LOG_ERR, "Failed to get peer_cfg");
+		return false;
+	}
+
+	child_cfg_t *child_cfg = NULL;
+	enumerator_t *enumerator = peer_cfg->create_child_cfg_enumerator(peer_cfg);
+	enumerator->enumerate(enumerator, &child_cfg);
+	if(! child_cfg) {
+		LOG(DBG_KNL, SL_LOG_ERR, "Failed to get child_cfg");
+		return false;
+	}
+
+	intervals->ike_rekey_interval = peer_cfg->get_rekey_time(peer_cfg, FALSE);
+	intervals->ike_reauth_interval = peer_cfg->get_reauth_time(peer_cfg, FALSE);
+	intervals->ipsec_rekey_interval = child_cfg->get_lifetime(child_cfg, FALSE)->time.life;
```

Comment:
There's a difference between the meaning of {rekey,reauth}-interval in operDB, and {rekey,reauth}-time here. The {rekey,reauth}-time are calcualted by subtracting the rekey_marking from the {rekey,reauth}-interval

Reply:
I don't see how this was resolved